### DISCUSSION TITLE
FTP 与防火墙有冲突，彻底关闭防火墙才能使用 FTP[BUG]

### DISCUSSION THREAD

**Discussion by Lockartist** [1Panel-dev/1Panel#5468] [Q&A]
### 联系方式

官方 9 号群，搜蓝冰

### 1Panel 版本

v1.10.10-lts

### 问题描述

防火墙开启状态，放行 21 端口的情况下，配置好的 ftp 无法连接。彻底关闭防火墙，马上就能连接。
<img width="722" alt="image" src="https://github.com/1Panel-dev/1Panel/assets/63927276/42b4160d-f6aa-4099-96e5-9277b1f75b0e">
<img width="1006" alt="image" src="https://github.com/1Panel-dev/1Panel/assets/63927276/39c8b740-1e15-451f-a9cf-76a6f347b32c">

### 重现步骤

开启防火墙，ftp就无法读取目录列表。关闭防火墙就 ok

**Comment by wanghe-fit2cloud** [comment]
要放行的端口包括：
1、控制连接端口（21）
2、被动模式数据端口范围（如30000到35000）
> 要配置被动模式端口范围，可以在 /etc/pure-ftpd/pure-ftpd.conf 文件中找到或添加以下行：
PassivePortRange 30000 35000

**Reply by Lockartist** [reply]
我试一下

**Reply by Lockartist** [reply]
<img width="1198" alt="image" src="https://github.com/1Panel-dev/1Panel/assets/63927276/a2270cc2-84f6-42a5-b3fb-813303816357">
<img width="449" alt="image" src="https://github.com/1Panel-dev/1Panel/assets/63927276/3aaa7343-c893-4dd3-9285-9830428d9693">
防火墙开启的情况下 21、30000-35000 都放行了也不行。

**Reply by wanghe-fit2cloud** [reply]
被动模式的端口范围不一定是 30000-35000，请自行查看配置文件以确定具体范围。

**Reply by Lockartist** [reply]
谢谢，查看配置文件后，被动端口是 30000-50000，放行后 ok 了。
该问题不是 bug，已经解决，希望官方在 1panel 上加个提示，或者在部署 ftp 的文档上，写上被动端口放行这个，适合我们这种小白·

**Reply by Lockartist** [reply]
不对，还是打不开。昨天开启防火墙后立马放行 30000-50000 ，连接 ftp，能连上。今天又连接不了了，是放行状态。防火墙关闭目录就读出来了，开启就不行·

**Comment by ssongliu** [comment]
ubuntu 被动端口配置不能直接修改配置文件，需要在 `/etc/pure-ftpd/conf `增加一个配置文件：
`echo '39000 40000' > /etc/pure-ftpd/conf/PassivePortRange `

根据实际需求调整端口范围，然后重启服务即可正常连接
